core/arch/arm/kernel/unwind_arm32.c

```diff
@@ -462,7 +463,7 @@ void print_kernel_stack(int level)
 	state.registers[FP] = read_fp();
 	state.registers[SP] = read_sp();
 	state.registers[LR] = read_lr();
-	state.registers[PC] = (uint32_t)print_kernel_stack;
+	state.registers[PC] = (uint32_t)print_kernel_stack + 4;
```

Comment:
Please add a comment describing this. Perhaps something like:
``` C
/*
 * Add 4 to make sure that we have an address well inside this function.
 * This is needed because we're subtracting 2 from PC when calling
 * find_index() above. See a comment there for more details.
 */

Reply:
How about using LR directly instead of plus 4 here?  For example:
state.registers[PC] = read_lr();

"PC plus 4" may move the PC to the next function of current one, if the current function has only one instruction...However, this is not the case for printk_kernel_stack, so "PC plus 4" still works.

But I think read_lr() may be more simple, as the current function itself is printk_kernel_stack(), and the lr should point to printk_kernel_stack + 4 in ARM32 case ,and printk_kernel_stack + 2 in thumb case.

Reply:
`read_lr()` is an inline function so we would get the old LR.

Reply:
got it, I will use "PC + 4" in my new push.  Thanks.